woocommerce/google-listings-and-ads
# Attribute Mapping - Integrate Modal API (Frontend)

### Changes proposed in this Pull Request:

Closes part of #1648 .

Implements the API for getting the sources and attributes in the Create Rule modal as well as mapping the names of the source in the table.


### Screenshots:


https://user-images.githubusercontent.com/5908855/192836685-c390ea16-b27b-496e-a894-e68920a50a57.mov




### Detailed test instructions:


1. Enter to Attend tribute Mapping and see how a placeholder table appears when the data is being loaded
2. After data is being loaded you see the right names in the Target columns. (Notice one empty because that attribute doesn't exists)
3.  Click on create modal. You see the modal open
4.  Click on the selector, you will see real Attributes  from the API
5. Select Adult
6. A spinner will appear until request is finished
7. See the enum selector appearing
8. Select Color attribute. Another spinner will appear
9. See how the field / fixed value selector appears
10. In the field selector you can see some of your Fields/Taxonomies listed
11. Select again Adult. No spinner/request runs. But the right  enum selector appears

### Additional details

⚠️  The test for Category selector will be added when API for gets ting categories is approved
⚠️  We are returning an object in sources. This Will be refactored soon.
